addons/resource/models/resource.py

```diff
@@ -333,65 +333,106 @@ def _check_attendance(self):
     # --------------------------------------------------
     # Computation API
     # --------------------------------------------------
+    # YTI TODO: Remove me in master
     def _attendance_intervals(self, start_dt, end_dt, resource=None, domain=None, tz=None):
+        if resource is None:
+            resource = self.env['resource.resource']
+        return self._attendance_intervals_batch(
+            start_dt, end_dt, resources=resource, domain=domain, tz=tz
+        )[resource.id]
+
+    def _attendance_intervals_batch(self, start_dt, end_dt, resources=None, domain=None, tz=None):
         """ Return the attendance intervals in the given datetime range.
             The returned intervals are expressed in specified tz or in the resource's timezone.
         """
+        self.ensure_one()
+        resources = self.env['resource.resource'] if not resources else resources
         assert start_dt.tzinfo and end_dt.tzinfo
         combine = datetime.combine
 
-        resource_ids = [resource.id, False] if resource else [False]
+        resources_list = list(resources) + [self.env['resource.resource']]
+        resource_ids = [r.id for r in resources_list]
```

Comment:
I don't get the point of having 3 potentially conflicting representations:
```python
resources # self.env['resource.resource'] recordset (empty or not)
resources_list # [recordset, ..., recordset, empty_recordset]
resource_ids  # [id, ..., id, False]
```
Please only use recordsets and ids if possible, mixing recordsets and list of recordsets is confusing (and error-prone, considering the ambiguous content of those variables).
```python
resources # self.env['resource.resource'] (length > -1)
resource_ids = resources.ids # [id, ..., id]
for resource in resources:
  ...
```

Reply:
@Feyensv It's due to the fact that we should allow empty recordsets in case we want the global attendances. 

Iterating on an empty recordset won't make any step in a for loop for example. That's the reason why I make a list of records. We faced the same issue in the mail module.